sys-apps/systemd/systemd-9999.ebuild

```diff
@@ -426,14 +426,15 @@ multilib_src_install_all() {
 	# Flatcar: Don't enable services in /etc, move to /usr.
 	systemd_enable_service multi-user.target systemd-networkd.service
 	systemd_enable_service multi-user.target systemd-resolved.service
+	systemd_enable_service multi-user.target remote-fs.target
```

Comment:
Some other files are also missing, I guess we need something like this (unless upstream deprecated it):
```
multi-user.target machines.target
multi-user.target remote-cryptsetup.target
network-online.target systemd-networkd-wait-online.service
sockets.target systemd-networkd.socket
```

Reply:
I was wondering whether to basically translate all the preset stuff from systemd into `system_enable_service`. For now I did that only for remote-fs. Can update it here too.

Reply:
I added `multi-user.target machines.target` and `multi-user.target remote-cryptsetup.target`. I also added `remount-fs.target systemd-pstore.service`. I don't know where did you get the other two - they are not listed in upstream's preset file.

Reply:
They come from the 2512.5.0 Stable release. This commit here c07b410caf58921a4d39f6b406c4ea8a3136514e has a list of CoreOS services.

Reply:
So `network-online.target systemd-networkd-wait-online.service` is not necessary I think, because there is a comment in `systemd-networkd.service` which we enable:

```
# We want to enable systemd-networkd-wait-online.service whenever this service
# is enabled. systemd-networkd-wait-online.service has
# WantedBy=network-online.target, so enabling it only has an effect if
# network-online.target itself is enabled or pulled in by some other unit.
Also=systemd-networkd-wait-online.service
```

Not sure about `sockets.target systemd-networkd.socket` - `systemd-networkd.socket` is pulled in by `systemd-networkd.service` also with the `Also=` option. So I guess it's not necessary anymore too.

Reply:
I checked and the `Also=` is not enough: after `sudo systemctl status network-online.target` I saw that `systemd-networkd-wait-online.service` is inactive while on 2512.2.0 it is active afterwards.

Reply:
Same situation for `systemd-networkd.socket` which gets activated on 2512.2.0 but not on 2605.5.0

Reply:
Added those, but I don't see any change, maybe it's something to investigate later:

systemd-networkd-wait-online.service:

```
core@localhost ~ $ systemctl status systemd-networkd-wait-online.service
● systemd-networkd-wait-online.service - Wait for Network to be Configured
     Loaded: loaded (/usr/lib/systemd/system/systemd-networkd-wait-online.service; disabled; vendor preset: disabled)
     Active: inactive (dead)
       Docs: man:systemd-networkd-wait-online.service(8)
```
systemd-networkd.socket:

```
core@localhost ~ $ systemctl status systemd-networkd.socket
● systemd-networkd.socket - Network Service Netlink Socket
     Loaded: loaded (/usr/lib/systemd/system/systemd-networkd.socket; disabled; vendor preset: disabled)
     Active: active (running) since Sat 2020-09-26 13:57:07 UTC; 7min ago
   Triggers: ● systemd-networkd.service
       Docs: man:systemd-networkd.service(8)
             man:rtnetlink(7)
     Listen: route 1361 (Netlink)
     CGroup: /system.slice/systemd-networkd.socket
```

systemd-networkd.service:

```
core@localhost ~ $ systemctl status systemd-networkd.service
● systemd-networkd.service - Network Service
     Loaded: loaded (/usr/lib/systemd/system/systemd-networkd.service; disabled; vendor preset: disabled)
     Active: active (running) since Sat 2020-09-26 13:57:09 UTC; 7min ago
TriggeredBy: ● systemd-networkd.socket
       Docs: man:systemd-networkd.service(8)
   Main PID: 729 (systemd-network)
     Status: "Processing requests..."
      Tasks: 1 (limit: 1119)
     Memory: 2.4M
     CGroup: /system.slice/systemd-networkd.service
             └─729 /usr/lib/systemd/systemd-networkd

Sep 26 13:57:09 localhost systemd[1]: Starting Network Service...
Sep 26 13:57:09 localhost systemd-networkd[729]: Enumeration completed
Sep 26 13:57:09 localhost systemd[1]: Started Network Service.
Sep 26 13:57:09 localhost systemd-networkd[729]: eth0: IPv6 successfully enabled
Sep 26 13:57:09 localhost systemd-networkd[729]: eth0: Link UP
Sep 26 13:57:09 localhost systemd-networkd[729]: eth0: Gained carrier
Sep 26 13:57:09 localhost systemd-networkd[729]: eth0: DHCPv4 address 10.0.2.15/24 via 10.0.2.2
Sep 26 13:57:09 localhost systemd-networkd[729]: eth0: Unable to shorten overlong DHCP hostname 'flatcar_production_qemu-2632-0-0', ignoring: Numerical argum>
Sep 26 13:57:11 localhost systemd-networkd[729]: eth0: Gained IPv6LL
```

Reply:
Have you tried pulling the target in before with `start network-online.target` or `sockets.target`?

Reply:
Anyway thanks, also for making the situation much more clear now than it was before.